_posts/2019-08-29-wfs3.md

```diff
@@ -0,0 +1,503 @@
+especially about the CRS parameter. Indeed, the parameter is now named
+`bbox-crs`{:.inline} and is defined according to the content of the
+`crs`{:.inline} property specified within the JSON collection. Moreover, you can
+filter by adding `field_name=value`{:.inline} directly in the request instead
+of using the `EXP_FILTER`{:.inline} parameter.
+
+{% capture summary %}`features.py`{:.inline}{% endcapture %}
+{% capture details %}
+``` python
+import requests
+import json
+
+bbox = "191617.5,6762287.0,209565.0,6772826.7"
+crs = "http://www.opengis.net/def/crs/EPSG/9.6.2/2154"
+filter = "denomination=menhir"
+http = ("http://qgisserver/wfs3/collections/Megaliths/items.json?"
+        "MAP=/tmp/project.qgz"
+        "&limit=3"
+        "&{}"
+        "&bbox={}"
+        "&bbox-crs={}"
+       .format(filter, bbox, crs))
+
+request = requests.get(http)
+
+if request.status_code == 200:
+  doc = json.loads(request.text)
+
+  print("Type: ", doc["type"])
+  print("Number matched: ", doc["numberMatched"])
+  print("Number returned: ", doc["numberReturned"])
+
+  for feature in doc["features"]:
+    print("Feature URL: ", feature["properties"]["url"])
+else:
+  print("Error")
+```
+{% endcapture %}
+<details>
+  <summary>{{ summary | markdownify | remove: '<p>' | remove: '</p>' }}</summary>
+  {{ details | markdownify }}
+</details>
+
+``` bash
+$ python features.py
+Type:  FeatureCollection
+Number matched:  7
+Number returned:  3
+Feature URL:  http://kartenn.region-bretagne.fr/ws/recensement/detail.php?id=divi23_195851
+Feature URL:  http://kartenn.region-bretagne.fr/ws/recensement/detail.php?id=dev17012017_234024
+Feature URL:  http://kartenn.region-bretagne.fr/ws/recensement/detail.php?id=divi23_200251
+```
+
+<br/>
+[![Pie Chart]({{ site.baseurl }}/images/posts/wfs3/menhir.jpg){:class="center-image" width="200px"}](http://kartenn.region-bretagne.fr/ws/recensement/detail.php?id=dev17012017_234024)
+<br/>
+
+<hr>
+
+## HTML template
+
+The beautiful thing with OpenAPI specification is the elegant simplicity,
+meaning that displaying a collection in HTML is pretty easy. This way,
+Alessandro has also implemented a default HTML template deployed with QGIS
+Server.
+
+Concretely, this implies that any user may explore the data serve by the server
+through a simple web page. For instance, the WFS3 landing page for QGIS Server
+is available at the next URL:
+`http://qgisserver/wfs3?MAP=/tmp/megaliths.qgz`{:.inline}.
+
+<br/>
+![OGC Badge]({{ site.baseurl }}/images/posts/wfs3/landing.png){:class="center-image" width="600px"}
+<br/>
+
+Then, you can explore the API by clicking on `API definition`{:.inline} or
+even displaying information on a specific layer thanks to the
+`Feature collections`{:.inline} link. By the way, we may note that the corresponding HTML
+page is accessible through the same request as the one for retrieving layers'
+capabilities, but `.json`{:.inline} has to be replaced by `.html`{:.inline}.
```

Comment:
Maybe add a note about content negotiation, if the endpoint is accessed through a browser, you will get HTML back even without explicitly adding `.html` postfix.

Reply:
Done